File: CalculadoradeNotas/app/src/main/java/com/devheverton/calculadoradenotas/MainActivity.java
package com.devheverton.calculadoradenotas;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private EditText n1,n2,n3,n4,numeroDeFaltas;
    private Button btnCalcular;
    private TextView txtResultado;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        n1 = findViewById(R.id.nota1);
        n2 = findViewById(R.id.nota2);
        n3 = findViewById(R.id.nota3);
        n4 = findViewById(R.id.nota4);
        numeroDeFaltas = findViewById(R.id.numeroDeFaltas);
        btnCalcular = findViewById(R.id.btnCalcular);
        txtResultado = findViewById(R.id.txtResultado);

        btnCalcular.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                int nota1 = Integer.parseInt(n1.getText().toString());
                int nota2 = Integer.parseInt(n2.getText().toString());
                int nota3 = Integer.parseInt(n3.getText().toString());
                int nota4 = Integer.parseInt(n4.getText().toString());
                int numerosDe_Faltas = Integer.parseInt(numeroDeFaltas.getText().toString());
                int media = (nota1 + nota2 + nota3 + nota4 ) /4;

if ( media >= 5 && numerosDe_Faltas <= 20){
    txtResultado.setText("Aluno foi Aprovado \n"+ "Média Final" + media);
    txtResultado.setTextColor(getColor(R.color.TEAL_3000));
}else if (numerosDe_Faltas>20){
    txtResultado.setText("Aluno foi Reporvado por Faltas \n" + "Media Final" + media);
    txtResultado.setTextColor(getColor(R.color.RED_3000));
}else if (media < 5){
    txtResultado.setText("Aluno foi Reporvado por Faltas \n" + "Media Final" + media);
    txtResultado.setTextColor(getColor(R.color.RED_3000));
}

            }
        });
    }
}

File: CalculadoradeNotas2/app/src/main/java/com/devheverton/calculadoradenotas/MainActivity.kt
package com.devheverton.calculadoradenotas


import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.devheverton.calculadoradenotas.R.color.green
import com.devheverton.calculadoradenotas.databinding.ActivityMainBinding


class MainActivity : AppCompatActivity() {
    private  lateinit var  binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.btCalcular.setOnClickListener {
            val nota1 = binding.editNota1.text.toString()
            val nota2 = binding.editNota2.text.toString()
            val nota3 = binding.editNota3.text.toString()
            val nota4 = binding.editNota4.text.toString()
            val numerosFaltas = binding.editFaltas.text.toString()

            if (nota1.isEmpty() || nota2.isEmpty() || nota3.isEmpty() || nota4.isEmpty() || numerosFaltas.isEmpty()){
              binding.txtResultado.setText("Preencha todos os campos!")
            binding.txtResultado.setTextColor(getColor(R.color.red))

        }else{
            calcularMedia(nota1.toInt() , nota2.toInt() , nota3.toInt() , nota4.toInt() , numerosFaltas.toInt())
            }

        }
    }


        fun calcularMedia(nota1:Int,nota2:Int, nota3:Int, nota4:Int, numerosFaltas:Int ) {
            val media = (nota1 + nota2 + nota3 + nota4) / 4
            if (media >= 5 && numerosFaltas <= 20) {
                binding.txtResultado.setText("Aluno foi Aprovado \n Média final: $media")
                binding.txtResultado.setTextColor(getColor(green))
            } else if (numerosFaltas >20) {
                binding.txtResultado.setText("Aluno foi Reprovado por Falta \n Média final: $media")
                binding.txtResultado.setTextColor(getColor(R.color.red))

            }else if (media <5)
                binding.txtResultado.setText("Aluno foi Reprovado por Nota \n Média final: $media")
            binding.txtResultado.setTextColor(getColor(R.color.red))

        }
    }



